### DISCUSSION TITLE
Backups option for only when players are in world?

### DISCUSSION THREAD

**Discussion by BackedUpBooty** [lloesche/valheim-server-docker#301] [Ideas]
First up, this is an awesome image, thank you for putting it together. It really does what I need. 

One thing I've seen on another image is an env variable that allows the server to only perform a backup if there are players, and it will pause backups when no one is on. I took a look through the 'default' list you have, I'm already using `RESTART_IF_IDLE` and `UPDATE_IF_IDLE` but I couldn't see one for backups. Does the variable `BACKUP_IF_IDLE` exist and can it be used? 

Again, thanks for the work!

**Comment by lloesche** [comment]
Good idea. We could add that function, but to be reliable it should create backups up to like 40 min. after the last player disconnected as the world is saved on a 20min timer and we would want a consistent `.old` file in our backup. Otherwise the backup might contain a corrupted world file and/or missing the last 20 min of building progress.

**Reply by BackedUpBooty** [reply]
Great points, yes I'd kind of wondered about that myself but didn't want to put too much into my first post :)

**Reply by lloesche** [reply]
I just pushed https://github.com/lloesche/valheim-server-docker/commit/6453ae6030107e2df7ee3ee2957d20f10225c792 which adds `BACKUPS_IF_IDLE` and `BACKUPS_IDLE_GRACE_PERIOD`.

See https://github.com/lloesche/valheim-server-docker#environment-variables and https://github.com/lloesche/valheim-server-docker#backups for details.

**Reply by BackedUpBooty** [reply]
Super awesome, thank you!
